src/core/server/integration_tests/saved_objects/migrations/group3/split_failed_to_clone.test.ts

```diff
@@ -0,0 +1,189 @@
+  type KibanaMigratorTestKit,
+} from '../kibana_migrator_test_kit';
+import { delay } from '../test_utils';
+import '../jest_matchers';
+import { CloneIndexParams } from '@kbn/core-saved-objects-migration-server-internal/src/actions';
+
+// mock clone_index from packages/core
+jest.mock('@kbn/core-saved-objects-migration-server-internal/src/actions/clone_index', () => {
+  const realModule = jest.requireActual(
+    '@kbn/core-saved-objects-migration-server-internal/src/actions/clone_index'
+  );
+  return {
+    ...realModule,
+    cloneIndex: (params: CloneIndexParams) => async () => {
+      // we need to slow down the clone operation for indices other than
+      // .kibana so that .kibana can completely finish the migration before we
+      // fail
+      if (params.target.includes('slow_clone'))
+        await new Promise((resolve) => setTimeout(resolve, 1000));
+      return realModule.cloneIndex(params)();
+    },
+  };
+});
+
+// define a type => index distribution
+const RELOCATE_TYPES: Record<string, string> = {
+  dashboard: '.kibana_slow_clone_1',
+  visualization: '.kibana_slow_clone_1',
+  'canvas-workpad': '.kibana_slow_clone_1',
+  search: '.kibana_slow_clone_2',
+  task: '.kibana_task_manager',
+  'epm-packages-assets': '.kibana_slow_clone_1',
+  // the remaining types will be forced to go to '.kibana',
+  // overriding `indexPattern: foo` defined in the registry
+};
+
+export const logFilePath = Path.join(__dirname, 'split_failed_to_clone.test.log');
+
+describe('when splitting .kibana into multiple indices and one clone fails', () => {
+  let esServer: TestElasticsearchUtils['es'];
+  let typeRegistry: ISavedObjectTypeRegistry;
+  let migratorTestKitFactory: () => Promise<KibanaMigratorTestKit>;
+
+  beforeAll(async () => {
+    typeRegistry = await getCurrentVersionTypeRegistry({ oss: false });
+    await clearLog(logFilePath);
+    esServer = await startElasticsearch({
+      dataArchive: Path.join(__dirname, '..', 'archives', '7.14.0_xpack_sample_saved_objects.zip'),
+      timeout: 60000,
+    });
+  });
+
+  afterAll(async () => {
+    await esServer?.stop();
+    await delay(2);
+  });
+
+  it('after resolving the problem and retrying the migration completes successfully', async () => {
+    const updatedTypeRegistry = overrideTypeRegistry(
+      typeRegistry,
+      (type: SavedObjectsType<any>) => {
+        return {
+          ...type,
+          indexPattern: RELOCATE_TYPES[type.name] ?? MAIN_SAVED_OBJECT_INDEX,
+        };
+      }
+    );
+
+    migratorTestKitFactory = () =>
+      getKibanaMigratorTestKit({
+        types: updatedTypeRegistry.getAllTypes(),
+        kibanaIndex: '.kibana',
+        logFilePath,
+        defaultIndexTypesMap: DEFAULT_INDEX_TYPES_MAP,
+      });
+
+    const { runMigrations: runMigrationsWhichFailsWhenCloning, client } =
+      await migratorTestKitFactory();
+
+    // count of types in the legacy index
+    expect(await getAggregatedTypesCount(client, '.kibana')).toEqual({
```

Comment:
nit, we can leverage jest, should result in slight better error reporting (same for further down)

```ts
await expect(getAggregatedTypesCount(client, '.kibana')).resolves.toEqual({
```